flask/helpers.py

```diff
@@ -534,8 +535,13 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
         if attachment_filename is None:
             raise TypeError('filename unavailable, required for '
                             'sending as attachment')
+        filename_dict = {
+            'filename': (unicodedata.normalize('NFKD',
+                         text_type(attachment_filename)).encode('ascii',
+                                                                'ignore')),
+            'filename*': "UTF-8''%s" % url_quote(attachment_filename)}
```

Comment:
Changing the Headers API to care about ordering of options would require a refactor. I don't think that is warranted for a workaround for older browsers.

We could check if the filename fits into ASCII and omit `filename*` accordingly. Then the user can decide whether they want to support IE8 or not.